### DISCUSSION TITLE
ShipmentMethod and ShipmentMethodName

### DISCUSSION THREAD

**Discussion by eric-makro** [saleor/saleor#10580] [Q&A]
Hi, Saleor users, I am new for Saleor platform.

I tried to create draft order with mutation draftOrderCreate and put shippingMethod correctly, but why it seems not detected from Saleor Order Draft dashboard, and shippingMethodName is null in db too. 
Please refer to below screenshot.

Can anyone please help how to create draft Order with ShipmentMethodName?

![iScreen Shoter - 25650908145721223](https://user-images.githubusercontent.com/110385557/189068977-8c12f33d-fcd9-47dd-ba79-029918f03515.jpg)

**Comment by peelar** [comment]
Hello @eric-makro 👋 

Can you please provide step-by-step instructions of what you did, as well as the content of your `draftOrderCreate` mutation?

**Reply by eric-makro** [reply]
Hi @peelar , thanks for your response.

Sure I will provide step-by-step in next reply.

**Reply by eric-makro** [reply]
Steps:
1. I run the app in my local.
2. I use `POST` to `localhost:8000/graphql/` 
3. Body:
```
mutation draftOrderCreate {
    draftOrderCreate(
        input: {
            billingAddress: {
                city: "Bangkok"
                country: TH
                streetAddress1: "billingstreet1"
            }
            shippingAddress: {
                city: "Bangkok"
                country: TH
                streetAddress1: "shippingstreet1"
            }
            shippingMethod: "U2hpcHBpbmdNZXRob2Q6MTE="
            lines: [
                {
                    quantity: 1,
                    variantId: "UHJvZHVjdFZhcmlhbnQ6MzYx"
                },
                {
                    quantity: 2,
                    variantId: "UHJvZHVjdFZhcmlhbnQ6Mzg1"
                }
                    ]
            channelId: "Q2hhbm5lbDox"
        }
    ) {
        order {
            id
            number
            shippingMethodName
            channel {
                id
            }
        }
        errors {
            field
            code
        }
        
    }
}
```
4. Response:
```
{
    "data": {
        "draftOrderCreate": {
            "order": {
                "id": "T3JkZXI6ZDVjODQ3YzEtNmRmYS00NjNmLWJlYjMtNTQ0OTM2OTk3NTI3",
                "number": "42",
                "shippingMethodName": null,
                "channel": {
                    "id": "Q2hhbm5lbDox"
                }
            },
            "errors": []
        }
    },
    "extensions": {
        "cost": {
            "requestedQueryCost": 1,
            "maximumAvailable": 50000
        }
    }
}
```
5. Saleor Dashboard
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/110385557/189087848-26982b0d-e215-46ac-b9ae-77737391b0fe.png">

Please kindly help, thank you

**Reply by patrys** [reply]
This looks like a bug, I'll convert this to an issue.

**Reply by eric-makro** [reply]
Thanks @patrys
